inlayHints.py

```diff
@@ -0,0 +1,91 @@
+from .protocol import InlayHint, InlayHintRequestParams, InlayHintResponse
+from html import escape as html_escape
+from LSP.plugin import Request
+from LSP.plugin import Session
+from LSP.plugin.core.protocol import Point
+from LSP.plugin.core.registry import windows
+from LSP.plugin.core.types import debounced
+from LSP.plugin.core.types import FEATURES_TIMEOUT
+from LSP.plugin.core.typing import List, Optional
+from LSP.plugin.core.views import point_to_offset
+from LSP.plugin.core.views import text_document_identifier
+import sublime
+import sublime_plugin
+
+
+def inlay_hint_to_phantom(view: sublime.View, hint: InlayHint) -> sublime.Phantom:
+    html = """
+    <body id="lsp-typescript-inlay-hints">
+        <style>
+            .inlay-hint {{
+                color: color(var(--foreground) alpha(0.6));
+                background-color: color(var(--foreground) alpha(0.08));
+                border-radius: 4px;
+                padding: 0.05em 4px;
+                font-size: 0.9em;
+            }}
+        </style>
+        <div class="inlay-hint">
+            {label}
+        </div>
+    </body>
+    """
+    point = Point.from_lsp(hint['position'])
+    region = sublime.Region(point_to_offset(point, view))
+    label = html_escape(hint["text"])
+    html = html.format(label=label)
+    return sublime.Phantom(region, html, sublime.LAYOUT_INLINE)
+
+
+def session_by_name(view: sublime.View, session_name: str) -> Optional[Session]:
+    listener = windows.listener_for_view(view)
+    if listener:
+        for sv in listener.session_views_async():
+            if sv.session.config.name == session_name:
+                return sv.session
+    return None
+
+
+class InlayHintsListener(sublime_plugin.ViewEventListener):
+    def __init__(self, view: sublime.View) -> None:
+        super().__init__(view)
+        self.phantom_set = sublime.PhantomSet(view, "_lsp_typescript_inlay_hints")
+
+    def on_modified_async(self) -> None:
+        change_count = self.view.change_count()
+        debounced(
+            self.request_inlay_hints_async,
+            FEATURES_TIMEOUT,
+            lambda: self.view.change_count() == change_count,
+            async_thread=True,
+        )
+
+    def on_load_async(self) -> None:
+        self.request_inlay_hints_async()
```

Comment:
Note that this might trigger before the session starts so then the file will not have annotations until file is re-activated or changed.

I would wish for tighter integration with LSP so that things are properly synced and cleared when session stops or is restarted.
But it would be non-trivial to do I guess.

Reply:
Storing the phantoms in SessionBuffer doesn't seem like a bad idea since that would properly clear the phantoms in some cases.

I only meant the "request on init" to be bad.